tiledb/sm/array_schema/enumeration.cc

```diff
@@ -86,19 +86,42 @@ Enumeration::Enumeration(
     throw EnumerationException("Invalid cell_val_num in Enumeration");
   }
 
-  if (data == nullptr || data_size == 0) {
-    throw EnumerationException("No attribute value data supplied.");
+  // Check input consistency. Non-nullptr inputs need a non-zero size. Nullptr
+  // inputs require a zero data_size.
+  if (data == nullptr && data_size != 0) {
+    throw EnumerationException(
+        "Invalid data input, nullptr with non-zero size.");
+  }
+
+  if (data != nullptr && data_size == 0) {
+    throw EnumerationException(
+        "Invalid data input, non-nullptr with zero size.");
   }
 
-  if (var_size() && (offsets == nullptr || offsets_size == 0)) {
+  if (offsets == nullptr && offsets_size != 0) {
+    throw EnumerationException(
+        "Invalid offsets input, nullptr with non-zero size.");
+  }
+
+  if (offsets != nullptr && offsets_size == 0) {
+    throw EnumerationException(
+        "Invalid offsets input, non-nullptr with zero size.");
+  }
+
+  auto data_provided = (data != nullptr && data_size > 0);
```

Comment:
Would it be possible to have only one string enumeration value that is an empty string? In this case we could have data == nullptr and/or data_size == 0...

Reply:
Oh darn it. Good catch! And I'm pretty sure that's the second time you've pointed that out and I've still haven't gotten that edge case committed to memory.

Reply:
Fixed. With tests.